api/model/api.go

```diff
@@ -1210,6 +1212,10 @@ type GraphQLProxyConfig struct {
 	RequestHeadersRewrite map[string]RequestHeadersRewriteConfig `json:"request_headers_rewrite,omitempty"`
 }
 
+type GraphQLProxyFeaturesConfig struct {
+	UseImmutableHeaders bool `json:"use_immutable_headers"`
```

Comment:
Similar to my comment above, if the feature is not required to be specified, let's make it optional.